**THIS REPO IS ARCHIVED, PLEASE GO TO [https://hbctraining.github.io/main](https://hbctraining.github.io/main) FOR CURRENT LESSONS.**

## Introduction to RNA-seq using high-performance computing (HPC)

| Audience | Computational Skills | Prerequisites | Duration |
:----------|:----------|:----------|:----------|
| Biologists | Beginner/Intermediate | None | 2-day workshop (~13 hours of trainer-led time)|

### Description

This repository has teaching materials for a 2-day Introduction to RNA-sequencing data analysis workshop. This workshop focuses on teaching basic computational skills to enable the effective use of an high-performance computing environment to implement an RNA-seq data analysis workflow. It includes an introduction to shell (bash) and shell scripting. In addition to running the RNA-seq workflow from FASTQ files to count data, the workshop covers best practice guidlelines for RNA-seq experimental design and data organization/management.

> These materials were developed for a trainer-led workshop, but are also amenable to self-guided learning.

### Learning Objectives

1.	Understand the necessity for, and use of, the command line interface (bash) and HPC for analyzing high-throughput sequencing data.
2.	Understand best practices for designing an RNA-seq experiment and analysis the resulting data.


### Contents

| Lessons            | Estimated Duration |
|:------------------------|:----------|
|[Introduction to the shell](lessons/01_the_filesystem.md) | 70 min |
|[Searching and redirection in shell](lessons/02_searching_files.md) | 45 min |
|[Shell scripts and `for` loops](lessons/03_loops_and_scripts.md) | 75 min |
|[Permissions and environment variables](lessons/04_permissions_and_environment_variables.md) | 50 min |
|[Project and data organization](lessons/05_data_organization.md) | 40 min |
|[RNA-seq experimental design best practices](lectures/rna-seq_design.pdf) | 50 min |
|[Introduction to High-Performance Computing for HMS-RC's Orchestra](lectures/HPC_intro_slides_Radhika.pdf) | 45 min |
|[RNA-seq data QC with FastQC](lessons/06_assessing_quality.md) | 75 min |
|[RNA-seq workflow: Alignment and Counting](lessons/07_rnaseq_workflow.md) | 90 min |
|[Automating the RNA-seq workflow](lessons/08_automating_workflow.md) | 60 min |
|[Alternative workflows for analyzing RNA-seq data](lectures/RNAseq-analysis-methods.pdf) | 15 min |
|[Quantifying expression using alignment-free methods (Salmon)](lessons/09_salmon.md) | 75 min |


### Dataset

The dataset used in this workshop can be downloaded [here]().


***
*These materials have been developed by members of the teaching team at the [Harvard Chan Bioinformatics Core (HBC)](http://bioinformatics.sph.harvard.edu/). These are open access materials distributed under the terms of the [Creative Commons Attribution license](https://creativecommons.org/licenses/by/4.0/) (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*

* *Some materials used in these lessons were derived from work that is Copyright Â© Data Carpentry (http://datacarpentry.org/). 
All Data Carpentry instructional material is made available under the [Creative Commons Attribution license](https://creativecommons.org/licenses/by/4.0/) (CC BY 4.0).*
